example/lib/mobile_scanner_overlay.dart

```diff
@@ -133,96 +132,199 @@
-// //             Text("QRCODE: $qr"),
-// //           ],
-// //         ),
-// //       ),
-// //       floatingActionButton: FloatingActionButton(
-// //           child: Text(
-// //             "press me",
-// //             textAlign: TextAlign.center,
-// //           ),
-// //           onPressed: () {
-// //             setState(() {
-// //               camState = !camState;
-// //             });
-// //           }),
-// //     );
-// //   }
-// // }
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+import 'package:mobile_scanner_example/barcode_scanner_window.dart';
+import 'package:mobile_scanner_example/scanner_error_widget.dart';
+
+void main() {
+  debugPaintSizeEnabled = false;
+  runApp(HomePage());
+}
+
+class HomePage extends StatefulWidget {
+  @override
+  HomeState createState() => HomeState();
+}
+
+class HomeState extends State<HomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(home: BarcodeScannerWithOverlay());
+  }
+}
+
+class BarcodeScannerWithOverlay extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<BarcodeScannerWithOverlay> {
+  String? qr;
+  String overlayText = "Please scan QR Code";
+  bool camState = false;
+  late BarcodeCapture currentBarcodeCapture ;
+  final MobileScannerController controller = MobileScannerController(
+    facing: CameraFacing.back,
+    autoStart: false
+  );
+  @override
+  initState() {
+    super.initState();
+  }
+
+  @override
+  dispose() {
+    super.dispose();
+    controller.dispose();
+  }
+
+  configureCameraSettings() {
+    setState(() {
+      camState = !camState;
+      controller.start();
+    });
+  }
+
+  onBarcodeDetect(BarcodeCapture barcodeCapture) {
+  setState(() {
+    currentBarcodeCapture = barcodeCapture;
+    overlayText = barcodeCapture.barcodes.last.displayValue!;
+  });
+  }
+
+
+
+  @override
+  Widget build(BuildContext context) {
+    final scanWindow = Rect.fromCenter(
+      center: MediaQuery.of(context).size.center(Offset.zero),
+      width: 200,
+      height: 200,
+    );
+    final height = MediaQuery.of(context).size.height;
+    final width = MediaQuery.of(context).size.width;
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Scanner with Overlay Example app'),
+      ),
+      body: Center(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Expanded(
+                child: camState
+                    ? Stack(
+                      fit: StackFit.expand,
+                      children: [
+                        Center(
+                        child: MobileScanner(
+                          fit: BoxFit.contain,
+                          onDetect: (BarcodeCapture barcodeCapture) => {
```

Comment:
Could you pass the function directly?

```suggestion
onDetect: onBarcodeDetect,
```